providerframework/metaservice/archiver/manager/manager_test.go

```diff
@@ -0,0 +1,168 @@
+	return 1, nil
+}
+
+func (f *fakeStruct) Close() error {
+	return nil
+}
+
+func (f *fakeStruct) Read(p []byte) (n int, err error) {
+	return 1, nil
+}
+
+type ManagerTestSuite struct {
+	suite.Suite
+	mgr          *archivalManager
+	archFileName string
+	typ          archiver.CompressionType
+	fwriter      *fakeStruct
+	fReader      *fakeStruct
+	invoke       func(Writer archiver.Writer) archiver.Writer
+	readInvoke   func(Reader archiver.Reader) archiver.Reader
+	archFilePath string
+}
+
+func (suite *ManagerTestSuite) BeforeTest(suiteName, testName string) {
+
+	switch testName {
+	case "TestGetArchiver":
+		suite.mgr = &archivalManager{
+			archiveYard: "fakeArchiveYard1",
+		}
+		suite.archFileName = "FakeArchfileName"
+
+		err := os.Mkdir("fakeArchiveYard1", 0777)
+		if err != nil {
+			fmt.Print(err)
+		}
+
+		suite.typ = archiver.CompressionType("fake")
+		suite.invoke = func(Writer archiver.Writer) archiver.Writer {
+			return suite.fwriter
+		}
+		RegisterCompressionWriterPlugins(suite.typ, suite.invoke)
+
+	case "TestGetArchiveReader":
+
+		suite.mgr = &archivalManager{
+			archiveYard: "fakeArchiveYard2",
+		}
+		err := os.Mkdir("fakeArchiveYard2", 0777)
+		if err != nil {
+			fmt.Print(err)
+		}
+		_, err = os.Create("fakeArchiveYard2/FakeArchfileName")
+		if err != nil {
+			fmt.Print(err)
+		}
+
+		suite.typ = archiver.CompressionType("name")
+
+		suite.fReader = &fakeStruct{}
+		suite.readInvoke = func(Reader archiver.Reader) archiver.Reader {
+			return suite.fReader
+		}
+		RegisterCompressionReaderPlugins(suite.typ, suite.readInvoke)
+
+		suite.archFilePath = "fakeArchiveYard2/FakeArchfileName"
+	}
+}
+
+func (suite *ManagerTestSuite) AfterTest(suiteName, testName string) {
+
+	switch testName {
+	case "TestGetArchiver":
+		err := os.Remove("fakeArchiveYard1/FakeArchfileName") //0755
+		if err != nil {
+			fmt.Print(err)
+		}
+		err = os.Remove("fakeArchiveYard1") //0755
+		if err != nil {
+			fmt.Print(err)
+		}
+
+	case "TestGetArchiveReader":
+		err := os.Remove("fakeArchiveYard2/FakeArchfileName") //0755
+		if err != nil {
+			fmt.Print(err)
+		}
+		err = os.Remove("fakeArchiveYard2") //0755
+		if err != nil {
+			fmt.Print(err)
+		}
+	}
+}
+
+func (suite *ManagerTestSuite) TestGetArchiver() {
+
+	//when the compression type is fake not registered
+	faketyp := "faketype"
+	_, _, err := suite.mgr.GetArchiver(archiver.CompressionType(faketyp), suite.archFileName)
+	expErr := fmt.Errorf("archival plugin[%s] not available", faketyp)
+	assert.Equal(suite.T(), expErr, err)
+
+	//success case
+	_, archFile, err := suite.mgr.GetArchiver(suite.typ, suite.archFileName)
+	assert.Nil(suite.T(), err)
+	assert.Equal(suite.T(), archFile, "fakeArchiveYard1/FakeArchfileName")
+
+	//when trying to give already existing file
+	fakefile := suite.archFileName
+	err1 := fmt.Errorf("archival file(fakeArchiveYard1/FakeArchfileName) already exist")
+	_, _, err = suite.mgr.GetArchiver(suite.typ, fakefile)
+	assert.Equal(suite.T(), err1, err)
```

Comment:
IMHO, Split the testcase based on the scenario handled. Please hanlde for all other test cases.

Reply:
Handled